vault/diagnose/tls_test.go

```diff
@@ -132,15 +87,183 @@ func TestTLSTrailingData(t *testing.T) {
+		t.Errorf("Bad error message: %s", err.Error())
+	}
+
+	listeners = []listenerutil.Listener{
+		{
+			Config: &configutil.Listener{
+				Type:                          "tcp",
+				Address:                       "127.0.0.1:443",
+				ClusterAddress:                "127.0.0.1:8201",
+				TLSCertFile:                   "./test-fixtures/ecdsa.crt",
+				TLSKeyFile:                    "./../../api/test-fixtures/keys/key.pem",
+				TLSClientCAFile:               "./../../api/test-fixtures/root/rootcacert.pem",
+				TLSMinVersion:                 "tls10",
+				TLSRequireAndVerifyClientCert: true,
+				TLSDisableClientCerts:         false,
+			},
+		},
+	}
+	err = ListenerChecks(listeners)
+	if err == nil {
+		t.Errorf("TLS Config check on fake certificate should fail")
+	}
+	if err.Error() != "tls: private key type does not match public key type" {
+		t.Errorf("Bad error message: %s", err.Error())
+	}
 }
 
-func TestTLSMismatchedCryptographicInfo(t *testing.T) {}
+// TestTLSMultiKeys verifies that a unique error message is thrown when a key is specified twice.
+func TestTLSMultiKeys(t *testing.T) {
+	listeners := []listenerutil.Listener{
+		{
+			Config: &configutil.Listener{
+				Type:                          "tcp",
+				Address:                       "127.0.0.1:443",
+				ClusterAddress:                "127.0.0.1:8201",
+				TLSCertFile:                   "./../../api/test-fixtures/keys/key.pem",
+				TLSKeyFile:                    "./../../api/test-fixtures/keys/key.pem",
+				TLSClientCAFile:               "./../../api/test-fixtures/root/rootcacert.pem",
+				TLSMinVersion:                 "tls10",
+				TLSRequireAndVerifyClientCert: true,
+				TLSDisableClientCerts:         false,
+			},
+		},
+	}
+	err := ListenerChecks(listeners)
+	if err == nil {
+		t.Errorf("TLS Config check on fake certificate should fail")
+	}
+	if err.Error() != "tls: failed to find certificate PEM data in certificate input, but did find a private key; PEM inputs may have been switched" {
+		t.Errorf("Bad error message: %s", err.Error())
+	}
+}
 
-func TestTLSContradictoryFlags(t *testing.T) {}
+// TestTLSMultiCerts verifies that a unique error message is thrown when a cert is specified twice.
+func TestTLSMultiCerts(t *testing.T) {
+	listeners := []listenerutil.Listener{
+		{
+			Config: &configutil.Listener{
+				Type:                          "tcp",
+				Address:                       "127.0.0.1:443",
+				ClusterAddress:                "127.0.0.1:8201",
+				TLSCertFile:                   "./../../api/test-fixtures/keys/cert.pem",
+				TLSKeyFile:                    "./../../api/test-fixtures/keys/cert.pem",
+				TLSClientCAFile:               "./../../api/test-fixtures/root/rootcacert.pem",
+				TLSMinVersion:                 "tls10",
+				TLSRequireAndVerifyClientCert: true,
+				TLSDisableClientCerts:         false,
+			},
+		},
+	}
+	err := ListenerChecks(listeners)
+	if err == nil {
+		t.Errorf("TLS Config check on fake certificate should fail")
```

Comment:
nit: `t.Error()`

Reply:
i'm going to stop commenting on all of these haha

Reply:
No worries! I'm going to resolve all of these but one, and make the change throughout this file wherever I'm not using formatting explicitly in the errors.